app/javascript/components/URIRef.jsx

```diff
@@ -0,0 +1,64 @@
+import React from "react"
+import PropTypes, { string } from "prop-types"
+
+/**
+ * Input component consisting of a simple textbox.
+ *
+ *  Sample Rails view usage:
+ *
+ * ```
+ * <%= react_component(:URIRef, { paramPrefix: 'example', name: 'title', value: { "@id": "http://example.com/vocab#bar"} }) %>
+ * ```
+ *
+ * When used in a form, this will submit the array `example[title][]`
+ * with a single value, `{value: "http://example.com/vocab#bar"}`.
+ */
+class URIRef extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: props.value,
+    };
+    this.handleTextChange = this.handleTextChange.bind(this);
+  }
+
+  handleTextChange(event) {
+    this.setState({value: { "@id": event.target.value} })
+  }
+
+  render () {
+    let textbox_name = `${this.props.paramPrefix}[${this.props.name}][][value]`
```

Comment:
@dsteelma-umd To align with https://issues.umd.edu/browse/LIBHYDRA-345 the last segment of the textbox name should be `[@id]`

Reply:
@peichman-umd  - Code has been updated.